coopaname_custom_contract/models/hr_contract.py

```diff
@@ -0,0 +1,20 @@
+from odoo import api, models
+
+
+class Contract(models.Model):
+    _inherit = "hr.contract"
+
+    @api.multi
+    def create_document(self):
+        self.ensure_one()
+        if self.type_id == self.env.ref(
+            "hr_cae_contract.hr_contract_type_cape"
+        ):
```

Comment:
don't you need to test on echelon as well?

Reply:
I think I reasoned `type_id` is already determined by `echelon`. But we could add it for safety/readability. Feel free to do so.